impl/src/main/java/jakarta/faces/component/html/HtmlComponentUtils.java

```diff
@@ -0,0 +1,41 @@
+package jakarta.faces.component.html;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import jakarta.el.ValueExpression;
+import jakarta.faces.component.UIComponent;
+
+class HtmlComponentUtils {
+
+    private static final String ATTRIBUTES_THAT_ARE_SET = "jakarta.faces.component.UIComponentBase.attributesThatAreSet";
+    private static final String OPTIMIZED_PACKAGE = "jakarta.faces.component.";
+
+    static void handleAttribute(UIComponent component, String name, Object value) {
+        final Map<String, Object> attributes = component.getAttributes();
+
+        @SuppressWarnings("unchecked")
+        List<String> setAttributes = (List<String>) attributes.get(ATTRIBUTES_THAT_ARE_SET);
+
+        if (setAttributes == null) {
+            String cname = component.getClass().getName();
+            if (cname != null && cname.startsWith(OPTIMIZED_PACKAGE)) {
```

Comment:
I realize this is original code but cname can really never be null.

Reply:
Good catch!